src/iqstat.cpp

```diff
@@ -363,3 +241,94 @@ void usage() {
     }
     printReleaseInfo( "iqstat" );
 }
+
+auto show_RuleExec( char *userName, 
+                    const char *ruleName, 
+                    int allFlag, 
+                    bool brief ) -> int
+{
+    genQueryInp_t genQueryInp{};
+    genQueryOut_t *genQueryOut{};
+
+    auto num_cols_selected = (brief ? once_instance.brief_format_len
+                                    : once_instance.long_format_len);
+
+    for (int i = 0; i < num_cols_selected; i++) {
+        addInxIval(&genQueryInp.selectInp,i1a[i],0);
+    }
+
+    if (!allFlag) {
+        snprintf( v1, BIG_STR, "='%s'", userName );
+        addInxVal(&genQueryInp.sqlCondInp,COL_RULE_EXEC_USER_NAME,v1);
+    }
+
+    std::string diagnostic {""};
+    if (ruleName != NULL && *ruleName != '\0') {
+        diagnostic = (boost::format( " (and matching key '%s')" ) % ruleName).str();
+        snprintf( v2, BIG_STR, "='%s'", ruleName );
+        addInxVal(&genQueryInp.sqlCondInp,COL_RULE_EXEC_ID,v2);
+    }
+
+    genQueryInp.maxRows = 50;
+    status = rcGenQuery( Conn, &genQueryInp, &genQueryOut );
+
+    if ( status == CAT_NO_ROWS_FOUND ) {
+        // Need to determine which table "no rows" refers to: USER or RULE.
+        namespace ia = irods::experimental::administration;
+        if (ia::client::exists(*Conn, ia::user{userName})) {
+            if ( allFlag ) {
+                printf( "No delayed rules pending%s\n", diagnostic.c_str() );
+            }
+            else {
+                printf( "No delayed rules pending for user %s%s\n", userName, diagnostic.c_str());
+            }
+            return 0;
+        }
+        else {
+            printf( "User %s does not exist.\n", userName );
+            return 0;
+        }
+    }
+
+    if (brief) {
+        printf( "id     name\n" );
+    }
+
+    auto increment_count_and_print_results = [&] {
+       printCount += printGenQueryResults( Conn, status, genQueryOut,
+                                           brief ? NULL: columnNames,
+                                           int{brief} );
+    };
+
+    increment_count_and_print_results();
+
+    while ( status == 0 && genQueryOut->continueInx > 0 ) {
+        genQueryInp.continueInx = genQueryOut->continueInx;
+        status = rcGenQuery( Conn, &genQueryInp, &genQueryOut );
+        if (!brief && genQueryOut->rowCnt > 0 ) {
+            printf( "----\n" );
+        }
+        increment_count_and_print_results();
+        clearGenQueryOut( genQueryOut );
+    }
+    clearGenQueryInp( &genQueryInp );
```

Comment:
Can the `genQueryOut` variable result in a leak if we never enter the loop?
Can it leak after exiting the loop?

Reply:
Looks like it.  Not a serious leak, but a `scoped_exit` thing might help.

Reply:
Well, now I  see how it can be done with at_scope_exit .
  - clearGenQueryOut is the problem. it's called at the end of the loop
  - it is not idempotent
  - i can't compensate in the lambda by setting genQueryOut to a nullptr b/c it's tested in the loop condition
 
I just don't think it's cleanly possible.  But I'm open to suggestions.

Reply:
One idea: possibly set / maintain a flag, like a ` gq_out_dirty` or similar name, to indicate whether the RAII should call  clearGenQueryOut.  It'd cloud the logic a tiny bit, but I suppose necessarily so, if we want to avoid the leak.

Reply:
Perhaps the clearing of the genQueryOut variable needs to be moved just before invoking rcGenQuery.

And you add another call to clear after exiting the loop.

Reply:
Seems to compile now. Will check as to function in the testing phase, which will be soon. May need to write at least one more test, I'm not sure the iqstat tests i've written are stressing it enough yet.... ; )